db/compaction/compaction_service_test.cc

```diff
@@ -906,7 +922,7 @@ TEST_F(CompactionServiceTest, TruncatedOutput) {
           ASSERT_OK(s);
           ASSERT_GT(file_size, 0);
 
-          ASSERT_OK(test::TruncateFile(env_, file_name, file_size / 2));
+          ASSERT_OK(test::TruncateFile(env_, file_name, file_size / 4));
```

Comment:
Assertion `assert(props.tail_start_offset <= file_size);` in CalculateTailSize() conditionally crash on cmake but not make in https://github.com/facebook/rocksdb/actions/runs/14273806153 on truncated file read. While I haven't figured out why, a way to mitigate is to truncate more aggressively so the assertion will definitely fail in make as well (verified). Then we also added the logic of skip calculating to prevent crashing.